xdrpp/cereal.h

```diff
@@ -51,6 +51,17 @@ load(Archive &ar, T &t)
   xdr_traits<T>::load(ar, t);
 }
 
+template<typename Archive, typename T>
+std::enable_if_t<xdr_traits<T>::is_container>
+save(Archive &ar, const T &t)
+{
+  // Do not call into xdr_traits<T>::save because it doesn't handle the size tag
```

Comment:
Can you expand on which problem you're trying to solve?

1. `make_size_tag` should only be used for variable size containers, and `is_container` is not the right trait
2. cereal's [specialization for std::vector](https://github.com/USCiLab/cereal/blob/02eace19a99ce3cd564ca4e379753d69af08c2c8/include/cereal/types/vector.hpp) seems to be doing what you're doing here? I don't see it "bypassing the archive adapter"
```
CEREAL_SAVE_FUNCTION_NAME( Archive & ar, std::vector<T, A> const & vector )
  {
    ar( make_size_tag( static_cast<size_type>(vector.size()) ) ); // number of elements
    for(auto && v : vector)
      ar( v );
  }
```

Reply:
(I'm asking because you don't seem to depend on name which as far as I can tell is the only use for `xdr::archive`)

Reply:
Looking at this code more closely, perhaps a cleaner solution would be to define a `cereal_override` for anything satisfying `xdr_traits<T>::is_container` instead of overloading `save`. We could easily do that in stellar-core instead. We've already taken this approach for anything satisfying `xdr_traits<T>::is_enum`. What do you think @MonsieurNicolas?